.github/workflows/pre-commit-test.yaml

```diff
@@ -0,0 +1,13 @@
+name: pre-merge-checks
+on:
+  pull_request:
+    branches: [main, develop]
+jobs:
+  Lint-Check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: npm install
+      - run: npm run lint-check
+      - run: npm run format-check
+      - run: npm test
```

Comment:
Are we not using `yarn` here?

Reply:
NO we are going with NPM here since no specifications were mentioned to use yarn.

Reply:
For what, the specifications are not mentioned.
If it's working with Cloudflare Workers, I can see they have command used in there docs

Reply:
By specifications I mean requirements. It wasn't mentioned as requirement to use YARN

Reply:
Any specific reason to use yarn specifically?

Reply:
`yarn` is comparatively better than `npm` in terms of speed and security. Reason being it parallely installs the dependencies while npm installs it `sequentially`. It also has some more advantages too.